concordium-node/src/consensus_ffi/ffi.rs

```diff
@@ -884,6 +1009,78 @@ impl ConsensusContainer {
 
         unsafe { stopImportingBlocks(consensus) }
     }
+
+    /// Look up the account in the given block.
+    /// The return value is a pair of the block hash which was used for the
+    /// query, and the protobuf serialized response.
+    ///
+    /// If the account cannot be found then a [tonic::Status::not_found] is
+    /// returned.
+    pub fn get_account_info_v2(
+        &self,
+        block_hash: &crate::grpc2::types::BlockHashInput,
+        account_identifier: &crate::grpc2::types::account_info_request::AccountIdentifier,
+    ) -> Result<([u8; 32], Vec<u8>), tonic::Status> {
+        use crate::grpc2::Require;
+        let (block_id_type, block_hash) =
+            crate::grpc2::types::block_hash_input_to_ffi(block_hash).require_owned()?;
+        let (acc_type, acc_id) =
+            crate::grpc2::types::account_identifier_to_ffi(account_identifier).require_owned()?;
+        let consensus = self.consensus.load(Ordering::SeqCst);
+        let mut out_data: Vec<u8> = Vec::new();
+        let mut out_hash = [0u8; 32];
+        let response: ConsensusQueryResponse = unsafe {
+            getAccountInfoV2(
+                consensus,
+                acc_type,
+                acc_id,
+                block_id_type,
+                block_hash,
+                out_hash.as_mut_ptr(),
+                &mut out_data,
+                copy_to_vec_callback,
+            )
+            .try_into()?
+        };
+        response.ensure_ok()?;
```

Comment:
Do we need to add some context here, or is the error already sufficient?

Reply:
I have added a bit of context, but not sure if it adds much. The status code "NotFound" is what ultimately should be used. Not the message string.

Reply:
And from the context of the query it should be clear what was not found.